Cargo.toml

```diff
@@ -8,18 +8,19 @@ license = "MIT OR Apache-2.0"
 edition = "2018"
 repository = "https://github.com/ZcashFoundation/ed25519-zebra"
 description = "Zcash-flavored Ed25519 for use in Zebra."
+resolver = "2"
 
 [package.metadata.docs.rs]
 features = ["nightly"]
 
 [dependencies]
-hex = "0.4"
-sha2 = "0.9"
+hex = { version = "0.4", default-features = false, features = ["alloc"] }
+sha2 = { version = "0.9", default-features = false }
 rand_core = "0.6"
-thiserror = "1"
-curve25519-dalek = "3"
+curve25519-dalek = { version = "3", default-features = false, features = ["alloc", "u64_backend"] }
 serde = { version = "1", optional = true, features = ["derive"] }
 zeroize = "1.2"
+hashbrown = "0.12.0"
```

Comment:
Comment: Basically the no_std version of the std HashMap: 

> Since Rust 1.36, this is now the HashMap implementation for the Rust standard library. However you may still want to use this crate instead since it works in environments without std, such as embedded systems and kernels.

https://crates.io/crates/hashbrown